### PR TITLE
Picturae import-- added importer build steps, other improvements to jenkins

### PR DESCRIPTION
**Pull request by delarocamcas**

### CODE REVIEW

File: image_client.py [hunk 1]
```diff
@@ -39,8 +39,8 @@ def __init__(self, config=None):
         ptc_timezone = timezone(timedelta(hours=-8), name="PST")
         self.datetime_now = datetime.now(ptc_timezone)
         self.update_time_delta()
-        if config is not None:
-            self.config = config
+        self.config = config
+        if config is not None and config.MAILING_LIST:
             self.monitoring_tools = MonitoringTools(config=config, report_path=config.REPORT_PATH)
```
Comment (foozleface) [inline image_client.py:44]: you have a conditional at the derived class level around the mailing list (botany importer and ich importer) and another one here - probably should have one or the other? Is there a case where we could get in trouble if we don't check both? Maybe some kind of cleverness around the config file where the image_client.py knows to check a vaiable named $COLLECTION_mailing_list (some variant on that) and we don't need the check in the derived class.

File: image_client.py [hunk 2]
```diff
@@ -162,10 +164,10 @@ def upload_to_image_server(self, full_path, redacted, collection, original_path=
...
             logging.info(f"Uploaded: {local_filename},{attach_loc},{url}")
             logging.info("adding to image")
-
-            self.monitoring_tools.add_imagepath_to_html(image_path=original_path,
-                                                        id=id,
-                                                        success=True)
+            if self.config.MAILING_LIST:
+                self.monitoring_tools.add_imagepath_to_html(image_path=original_path,
+                                                            id=id,
+                                                            success=True)
```
Comment (foozleface) [inline image_client.py:171]: This whole block (lines 130-170) seem a bit convoluted. Without any thought at all, below  what chatGPT output for that code block, revised. I'm not advocating it (haven't even proofed it) but I think some kind of restructuing is in order.
if (
    "undatabased" in original_path.lower()
    and self.monitoring_tools.path != self.config.ACTIVE_REPORT_PATH
    and self.config.MAILING_LIST
):
    self.monitoring_tools.path = self.config.ACTIVE_REPORT_PATH

if id is None:
    id = "N/A"

if r.status_code != 200:
    self.logger.debug(f"FAIL - return code {r.status_code}. data: {data}")

    if r.status_code == 409:
        self.logger.error(f"Image already in server; skipping for {upload_path}")
        raise DuplicateImageException
    else:
        self.logger.error(f"Image upload aborted: {r.status_code}:{r.text}")

    if self.config.MAILING_LIST:
        self.monitoring_tools.add_imagepath_to_html(
            image_path=original_path,
            id=id,
            success=False,
        )
    raise UploadFailureException

params = {
    # ... other params ...
}

logging.info(f"Uploaded: {local_filename},{attach_loc},{url}")
logging.info("adding to image")

if self.config.MAILING_LIST:
    self.monitoring_tools.add_imagepath_to_html(
        image_path=original_path,
        id=id,
        success=True,
    )

File: importer_jenkins.template.sh
Comment (foozleface) [inline importer_jenkins.template.sh:1]: Maybe break this up into multiple shell files called by a master one? this is fairly long, and it looks like a bunch of steps. That might make debugging and reading easier. your call.

File: iz_import.sh [hunk 3]
```diff
@@ -20,4 +20,12 @@ else
     cd "$(dirname "$0")" || exit
     #source ./env/bin/activate
     python3 ./client_tools.py -vvvv IZ import >& iz_import_log.txt &
```
Comment (foozleface) [inline iz_import.sh:22]: probably need the source venv/bin/activate like the other two?

File: picturae_importer.py [hunk 4]
```diff
@@ -91,11 +91,11 @@ def init_all_vars(self):
...
-
-        self.monitoring_tools = MonitoringToolsDir(config=self.picturae_config,
-                                                   batch_md5=self.batch_md5,
-                                                   report_path=self.picturae_config.ACTIVE_REPORT_PATH,
-                                                   active=True)
+        if self.picturae_config.MAILING_LIST:
+            self.monitoring_tools = MonitoringToolsDir(config=self.picturae_config,
+                                                       batch_md5=self.batch_md5,
+                                                       report_path=self.picturae_config.ACTIVE_REPORT_PATH,
+                                                       active=True)
```
Comment (foozleface) [inline picturae_importer.py:99]: Yeah, we have four of these code blocks, If we can push all this logic down a level everything cleans up nicely.

**Review by foozleface — CHANGES_REQUESTED** [review]
The only important changes is the venv in iz_import and pushing the mailing list logic down a level. Other changes at your discretion.

File: importer_jenkins.template.sh
Comment (delarocamcas) [reply importer_jenkins.template.sh:1]: Sure thing, I can split it into the intitial unittest section, and then the have each importer run in a second .sh script

File: ichthyology_importer.py [hunk 5]
```diff
@@ -77,10 +77,8 @@ def get_catalog_number(self, filename):
         return f'{number}', collection
 
     def build_filename_map(self, full_path):
```
Comment (delarocamcas) [inline ichthyology_importer.py:79]: full_path=full_path.lower() , some ubuntu systems are case sensitive thus this errored out in the test env. , I removed it and did a test import, but I might want to double check that taking this out doesn't throw a wrench in any functions down the line.

File: Jenkinsfile
Comment (delarocamcas) [inline Jenkinsfile:1]: using workspace variable instead of regex

File: PIC_dbcreate/Picturae_DDL.sql
Comment (delarocamcas) [inline PIC_dbcreate/Picturae_DDL.sql:1]: forgot to add this table in on prior PRs , necessary for notes from nature expedition assembly

File: image_client.py [hunk 6]
```diff
@@ -39,9 +39,21 @@ def __init__(self, config=None):
         ptc_timezone = timezone(timedelta(hours=-8), name="PST")
         self.datetime_now = datetime.now(ptc_timezone)
         self.update_time_delta()
-        if config is not None:
-            self.config = config
-            self.monitoring_tools = MonitoringTools(config=config, report_path=config.REPORT_PATH)
+        self.config = config
+
```
Comment (delarocamcas) [inline image_client.py:43]: initializing monitoring tools with image_client

**Pull request merged**